In [1]:
# Плотность газа при различных давлениях и температуре:
def gas_density(M, t, p):
  # Температуру следует вводить в градусах Цельсия, давление - в атм.
  # Результат в кг/м3
  return (M * 273 * p) / (273 + t)

# 1 атм = 760 мм рт. ст., 101,3 КПа,  1,033 кгс/см2
# 1 мм рт. ст. = 133.4 Па

#Молекулярные массы:
M = {"CO2" : 44,
     "N2" : 28,
     "H2O" : 18,
     "H2" : 2,
     "CO" : 28
     }

# Основное уравнение гидростатики:
def pressure_static(rho, h):
  # Плотность и высота должны быть приведены в единицах СИ
  return rho*h*9.81

1.1 Найти мольную массу и лотность водяного газа при t = 90 °C и $p_{абс}$ = 1,2 $кгс/см^2$ (~0.12 МПа). Состав водяного газа: $H_2$  - 50 %, $CO$ - 40 %, $N_2$ - 5 %, $CO_2$ - 5 % (по объему).

In [2]:
# Давление нужно перевести в абсолютное:
p = 1 + 200 / 101.3
density = gas_density(44, 85, p)
print(f"{density:0.2f} кг/м3")

99.80 кг/м3


1.3 Состав продуктов горения 1 кг коксового газа (кг): $CO_2$ - 1.45, $N_2$ - 8,74, $H_2O$ - 1.92 %. Найти объемный состав продуктов горения.

In [3]:
# Состав продуктов горения массовый:
components_mass = {
    "CO2" : 1.45,
    "N2" : 8.74,
    "H2O" : 1.92
}

#Объемный состав продуктов горения:
components_volumes = {k:v / M[k] for k, v in components_mass.items()}
total_volume = sum(components_volumes.values())
components_volumes = {k:v/total_volume for k, v in components_volumes.items()}
for k, v in components_volumes.items():
  print(f"Объемная доля {k} - {v:0.3f}")

Объемная доля CO2 - 0.073
Объемная доля N2 - 0.691
Объемная доля H2O - 0.236


1.4 Разрежение в осушительной башне сернокислотного завода измеряется U-образным тягомером, наполненным серной кислотой плотностью 1800 кг/$см^3$. Показания тегомера 3 см. Каково асболютное давление в башне, выраженное в Па, есл барометрическое давление составялет 750 мм рт. ст.?

In [4]:
pressure_change = pressure_static(1800, 0.03)
pressure_absolute = 750 * 133.4 - pressure_change
print(f"Абсолютное давление в осушительной башне: {pressure_absolute:0.2f} Па")

Абсолютное давление в осушительной башне: 99520.26 Па


1.5 Манометр на трубопроводе, заполненном жидкостью, показывает давление 0,18 кгс/$см^2$. На какую высоту h над точкой присоединения манометра поднимается в открытом пьезометре жидкость, находящаяся в трубопроводе, если эта жидкость: а) вода, б) четыреххлористый углерод.

**Решение:**
Связь давления с высотой подъема жидкости:  

$\rho \cdot g \cdot h = p [Па]$

Следует учесть перевод единиц

$p[Па] = p[Н/м^2] = 10^4 \cdot g \cdot p[кгс/см^2]$

Высота подъема:

$h = 10^4 \cdot p[кгс/см^2] / \rho$

In [5]:
height_rho = 10**4 * 0.18
print(f"Высота подъема воды: {height_rho / 1000:0.2f} м")
print(f"Высота подъема ЧХУ: {height_rho / 1594:0.2f} м")

Высота подъема воды: 1.80 м
Высота подъема ЧХУ: 1.13 м


1.6 Высота уровня мазута в резервуаре 7.6 м. Относительная плотность мазута 0,96. На высоте 800 мм от дна в резервуаре имеется круглый лаз диаметров 760 мм, крышка которого прикрепляется болтами диаметров 10 мм. Принимая для болтов допустимое напряжение на разрыв 700 $кгс/см^2$, определить необходимое число болтов. Определить также давление мазута на дно резервуара.

In [6]:
import math

def hydrostatic_pressure_kgs(rho, h):
  return rho * h / 10**4

print(f"Давление на дно резервуара: {hydrostatic_pressure_kgs(960, 7.6) :0.3f} кгс/см2")

midle_window = 7.6 - 0.8
middle_window_pressure = hydrostatic_pressure_kgs(960, midle_window)
print(f"Среднее давление на крышку: {middle_window_pressure:0.3f} кгс/см^2")

window_area_cm2 = 3.1415 * 76**2 / 4
window_pressure_force = window_area_cm2 * middle_window_pressure
print(f"Сила действующая на крышку: {window_pressure_force:0.1f} кгс")

one_bolt = 700 * 3.1415 * 1 **2 / 4
print(f"Сила, выдерживаемая одним болтом: {one_bolt:0.1f} кгс")

n_bolts = math.ceil(window_pressure_force / one_bolt)
print(f"Требуемое число болтов: {n_bolts:0.0f} штук")

Давление на дно резервуара: 0.730 кгс/см2
Среднее давление на крышку: 0.653 кгс/см^2
Сила действующая на крышку: 2961.3 кгс
Сила, выдерживаемая одним болтом: 549.8 кгс
Требуемое число болтов: 6 штук


1.7 На малый поршень диаметром 40 мм ручного гидравлическкого пресса действует сила 589 Н (60 кгс). Перебрегая потерями, определить силу, действующую на прессуемое тело, если диаметр большого поршня 300 мм.

In [7]:
def hydravlic_press(d_small, d_large, force):
  ratio = (d_large / d_small) ** 2
  return force * ratio

print(f"Сила действующая на прессуемое тело: {hydravlic_press(40, 300, 589):0.0f} Н, {hydravlic_press(40, 300, 60):0.0f} кгс")

Сила действующая на прессуемое тело: 33131 Н, 3375 кгс


1.8 Динамический коэффициент вязкости жидкости при 50 °С равняется 30 мПа*с. Относительная плотность жидкости 0,9. Определить кинематический коэффициент взкости.

In [8]:
print(f"Кинематическая вязкость: {30e-3 / 900:0.2e} м3/с")

Кинематическая вязкость: 3.33e-05 м3/с


1.9 Найти динамический коэффициент вязкости ри 20 °С и атмосферном давлении азотоводородной смеси, водержащей 75 % водорода и 25 % азота (по объему).

In [9]:
# Зависимость вязкости от давления:
def gase_viscosity_temp(mu, T, C=100):
  return mu * (((273 + C) / (T + C)) * (T / 273)**1.5)

def gase_mixture_viscosity(viscosities, volumes, masses):
  n = len(viscosities)
  mixture_mass = 0
  adjusted_viscs = 0
  for i in range(n):
    partial_mass = volumes[i] * masses[i]
    mixture_mass += partial_mass
    adjusted_viscs += partial_mass / viscosities[i]

  return mixture_mass / adjusted_viscs

In [10]:
viscosities_0 = [17e-6, 8.42e-6]
C = [114, 73]

viscosities = [gase_viscosity_temp(v, 273+20, c) for v, c in zip(viscosities_0, C)]
print(f"Вязкость при заданной температуре азота - {viscosities[0]:0.2e}, водорода - {viscosities[1]:0.2e} Па*с")

print(f"Вязкость газовой смеси {gase_mixture_viscosity(viscosities, [0.25, 0.75], [28, 2]):0.2e} Па*с")

Вязкость при заданной температуре азота - 1.80e-05, водорода - 8.85e-06 Па*с
Вязкость газовой смеси 1.52e-05 Па*с


1.10 Известно, что динамический коэффициент взкости льняного масла при 30 °С равняется 0.331 П, а при 50°С - 0,176 П. Чему будет равен динамический коэффициент вязкости этого масла при 90°С? (Воспользоваться правилом линейности, приняв за стандартную жидкость, например 100 %-ный глицерин).

1.11 Холодильник состоит из 19 труб диаметров 20х2 мм. В трубное пространство холодильника поступает вода по трубопроводу диаметров 57х3.5 мм. Скорость воды в трубопроводе 1,4 м/с. Вода идет снизу вверх. Определить скорость воды в трубах холодильника.

In [11]:
#Из неразрывности потока следует, что расход воды в теплообменнике должен быть
#равен расходу воды в трубопроводе.
# Следовательно отношение скоростей будет определяться отношением сечений трубопроводов

S_ratio = (57 - 3.5 * 2) ** 2 / (19 * (20 - 2 * 2) ** 2)
print(f"Отношение сечения трубопровода суммарному сечению труб в теплообменнике: {S_ratio:0.3f}")

print(f"Скорость воды в трубах теплообъменника: {1.4 * S_ratio : 0.3f} м/с")

Отношение сечения трубопровода суммарному сечению труб в теплообменнике: 0.514
Скорость воды в трубах теплообъменника:  0.720 м/с


1.12 По трубам теплообменника, состоящего из 379 труб диаметром 16х1,5 мм, проходит азот в количестве $6400  м^3/ч$ (считая при 0 °C и 760 мм рт. ст.) под давлением $p_{изб} = 3 кгс/см^2 (~ 0.3 МПа)$. Азот входит в теплообменник при температуре 120 °C, выходит при 30 °C. Определить скорость азота в трубах теплообменника на входе и на выходе.

In [12]:
# Сделаем функцию для пересчета объема газа от нормальных условий до требуемых:

def gas_volume(normal_volume, pressure, temperature, normal_pressure = 1, normal_temperature = 0):
  return normal_volume * (273 + temperature) * normal_pressure / (pressure * (273 + normal_temperature))

# Сечение труб теплообменника:
S = 379 * 3.1415 * (0.016 - 2 * 0.0015 ) ** 2 / 4
print(f"Сечение труб теплообменника: {S:0.2e} м2")

# Расход азота:
N2_v_in = gas_volume(6400, (100000 + 3*98100), 120, normal_pressure=133.4*760)
N2_v_out = gas_volume(6400, (100000 + 3*98100), 30, normal_pressure=133.4*760)
print(f"Расход азота на входе {N2_v_in:0.2e} м3/ч")
print(f"Расход азота на выходе {N2_v_out:0.2e} м3/ч")

# Скорость азота:
print(f"Скорость азота на входе: {N2_v_in /(3600 * S) :0.2f} м/с")
print(f"Скорость азота на выходе: {N2_v_out /(3600 * S) :0.2f} м/с")

Сечение труб теплообменника: 5.03e-02 м2
Расход азота на входе 2.37e+03 м3/ч
Расход азота на выходе 1.83e+03 м3/ч
Скорость азота на входе: 13.08 м/с
Скорость азота на выходе: 10.09 м/с


1.13 Холодильник состоит из двух труб из двух концентрических стальных труб диаметром 29х2,5 мм и 54х2,5 мм. По внутенней трубе протекают 3,73 т/ч рассола плотностью 1150 $кг / м^3$. В межтрбном пространстве проходит 160 кг/ч газа под давлением $p_{абс} = 3 кгс/см^2$ (~0.3 МПа) при средней температуре 0 °C. Плотность газа при 0 °C  и 760 мм рт. ст. равна 1,2 $кг/м^3$. Найти скорости газа и жидкости в холодильнике.

In [13]:
# Площади поперечного сечения труб:
outer = 3.1415 * ((0.054 - 0.005)**2 - 0.029**2) / 4
inner = 3.1415 * (0.029 - 0.005)**2 / 4
print(f"Площадь сечение внутренней трубы: {inner:0.2e} м2")
print(f"Площадь сечение внешней трубы: {outer:0.2e} м2")
print()

# Расход жидкости
vol_l = 3730 / (3600 * 1150)
print(f"Расход рассола: {vol_l:0.2e} м3/c")
vel_l = vol_l / inner
print(f"Скорость рассола: {vel_l:0.2f} м/c")
print()

# Расход газа
vol_g = (760 * 133.4 / (3e4 * 9.81)) * 160 / (3600 * 1.2)

print(f"Расход газа: {vol_g:0.2e} м3/c")
vel_g = vol_g / outer
print(f"Скорость газа: {vel_g:0.2f} м/c")
print()

Площадь сечение внутренней трубы: 4.52e-04 м2
Площадь сечение внешней трубы: 1.23e-03 м2

Расход рассола: 9.01e-04 м3/c
Скорость рассола: 1.99 м/c

Расход газа: 1.28e-02 м3/c
Скорость газа: 10.41 м/c



1.14 Определить необходимый диаметр наружной трубы в условиях предыдущей задачи, если газ пойдет под атмосферным давлением, но при той же скорости и при том же массовом расходе.

In [14]:
vol_g =  160 / (3600 * 1.2)

print(f"Расход газа: {vol_g:0.2e} м3/c")

outer_new = vol_g / vel_g
print(f"Площадь сечения внешней трубы: {outer_new:0.2e} м2")

outer_d = (outer_new * 4 / 3.1415 + 0.029**2) ** 0.5
print(f"Внутренний диаметр внешней трубы: {outer_d*1000:0.0f} мм")

Расход газа: 3.70e-02 м3/c
Площадь сечения внешней трубы: 3.56e-03 м2
Внутренний диаметр внешней трубы: 73 мм
